docs/Changelog.md

```diff
@@ -18587,6 +18587,95 @@ This version of the operator has been available since version 13 of the default
 </dl>
 
 ## Version 14 of the default ONNX operator set
+### <a name="BatchNormalization-14"></a>**BatchNormalization-14**</a>
+
+  Carries out batch normalization as described in the paper
+  https://arxiv.org/abs/1502.03167. Depending on the mode it is being run,
+  There are five required inputs 'X', 'scale', 'B', 'input_mean' and
+  'input_var', in addition to one optional input 'training_mode'.
+  Note that 'input_mean' and 'input_var' are expected to be the estimated
+  statistics in inference mode (training_mode=False, default),
+  and the running statistics in training mode (training_mode=True).
+  There are multiple cases for the number of outputs, which we list below:
+  
+  Output case #1: Y, running_mean, running_var, current_mean, current_var (training_mode=True)
+  Output case #2: Y (training_mode=False)
+  
+  When training_mode=False, extra outputs are undefined and the user should not depend on those.
+  The outputs are updated as follows when training_mode=True:
+  ```
+  current_mean = ReducedMean(X, axis=all_except_channel_index)
```

Comment:
Do you mean `ReduceMean` instead of `ReducedMean`? (same for var)

Otherwise the formulas look good to me (I especially like the use of `running_mean` which is less confusing than the `saved_mean` that the older spec had).

Reply:
Thanks. I can update this.
I updated the part for optional outputs, mentioning this is invalid. What do you think?

Reply:
Nice! I think in addition the shape inference could be updated to enforce that no optional outputs are present if `training_mode=true`

Reply:
Sure. Will do. 
@pranav-prakash Do you think you can help us test this spec with you ORT implementation? Just to check spec works fine.

Reply:
Sure, I can do so (next week though, I'm a bit busy this week). Even with the change in spec I don't there would be be any changes in my  PR for ORT BatchNorm training though (aside from checking the newly-added `training_mode` attribute).

 (There's also a separate issue where the pre-existing CUDA BatchNorm training implementation in ORT deviates from the spec in outputting inverse-standard-deviation instead of variance – whose behavior I copied so the tests could remain the same – but that's a discussion for that PR).